ReactNativeClient/lib/services/ResourceFetcher.js

```diff
@@ -206,25 +207,31 @@ class ResourceFetcher extends BaseService {
 	}
 
 	async autoAddResources(limit = null) {
-		if (limit === null) limit = 10;
+		this.autoAddResourcesCalls_.push(true);
+		try {
+			if (limit === null) limit = 10;
 
-		if (this.addingResources_) return;
-		this.addingResources_ = true;
+			if (this.addingResources_) return;
```

Comment:
If we can return here, doesn't that mean that there will be a memory leak in your autoAddResourcesCalls_ array? (unless I missed something but the diff again is hard to follow)

Reply:
No, I don't think so.  When we return within the `try` block, the `finally` will run, and so no memory leak.